File: 08-multi-project/02-multi-project-springboot/webserver/src/main/java/org/pkaq/Booter.java
package org.pkaq;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * @Description:     应用配置启动入口.
 * @FileName:         Booter.java
 * @Author:            S.PKAQ
 * @Version:          1.0
 * @Date:              2016年4月7日 上午9:28:07
 */
@SpringBootApplication
@Slf4j
public class Booter implements CommandLineRunner {
    /**
     * 入口函数.
     * @param args args
     */
    @Autowired
    public static void main(String[] args) {
        SpringApplication.run(Booter.class, args);
    }

    public void run(String... args) throws Exception {
        log.info("  --- --- --- [ web started ] --- --- ---  ");
    }
}

File: 14-kotlin/00-spring_boot/src/main/kotlin/org/pkaq/Booter.kt
package org.pkaq

import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping

@SpringBootApplication
@Controller
open class Booter:CommandLineRunner{
    @Throws(Exception::class)
    override fun run(vararg args: String) {
        println("  --- --- --- [ web started ] --- --- ---  ")
    }

    @RequestMapping("/")
    fun index():String {
        return "index"
    }
}

fun main(args: Array<String>) {
    org.springframework.boot.SpringApplication.run(org.pkaq.Booter::class.java, *args)
}